lib/pure/uri.nim

```diff
@@ -59,7 +60,14 @@ proc parseAuthority(authority: string, result: var Uri) =
       result.hostname.setLen(0)
       inPort = false
     of ':':
-      inPort = true
+      if inIPv6:
+        result.hostname.add(authority[i])
+      else:
+        inPort = true
+    of '[':
+      inIPv6 = true
+    of ']':
+      inIPv6 = false
     of '\0': break
     else:
       if inPort:
```

Comment:
This whole parseAuthority proc is bad. It will consider many invalid URIs as valid and will parse them meaninglessly. It should be changed so that it robustly parses the authority according to the grammar specified in RFC (same applies to other "parseSomething" procs in the module).